server/controllers/contest.js

```diff
@@ -60,3 +62,31 @@ exports.getContests = asyncHandler(async (req, res, next) => {
     throw new Error('failed to get contests');
   }
 });
+
+// handler for create submission by contest id
+exports.createSubmissionByContestId = asyncHandler(async (req, res, next) => {
+  const contestId = req.params.id;
+  const s3UrlArray = req.body.data;
+  const userId = req.user.id;
+  if (!ObjectId.isValid(userId)) {
```

Comment:
We don't need to check if user id is valid as it persists in the jwt token for a logged in user. However, I guess you can check the contest id for validity :)

Reply:
ok, changed to check contest id